modules/okhttp/src/main/scala/coursier/cache/protocol/HttpHandler.scala

```diff
@@ -1,17 +1,51 @@
 package coursier.cache.protocol
 
 import java.net.{URLStreamHandler, URLStreamHandlerFactory}
+import java.util.concurrent.TimeUnit
 
 import com.squareup.okhttp.{OkHttpClient, OkUrlFactory}
 
+import scala.concurrent.duration.FiniteDuration
+
+/**
+  * Contains the default values for the OK Http client.
+  *
+  * The default client can be configured using properties which are first checked from the system properties, then environment variables:
+  *
+  * - COURSIER_CONN_TIMEOUT_SECONDS : the http connection timeout, in seconds
+  * - COURSIER_READ_TIMEOUT_SECONDS : the http read timeout, in seconds
+  *
+  */
 object HttpHandler {
-  lazy val okHttpClient = new OkHttpClient
-  lazy val okHttpFactory = new OkUrlFactory(okHttpClient)
+
+  private def durationInSecondsFromEnv(key : String): Option[FiniteDuration] = {
+    import concurrent.duration._
+    sys.props.get(key).orElse(sys.env.get(key)).map(_.toInt.seconds)
+  }
+
+  def newClient() = {
+    val client = new OkHttpClient
+
+    durationInSecondsFromEnv("COURSIER_CONN_TIMEOUT_SECONDS").foreach { timeoutSeconds =>
+      client.setConnectTimeout(timeoutSeconds.toSeconds, TimeUnit.SECONDS)
+    }
+
+    durationInSecondsFromEnv("COURSIER_READ_TIMEOUT_SECONDS").foreach { timeoutSeconds =>
+      client.setReadTimeout(timeoutSeconds.toSeconds, TimeUnit.SECONDS)
+    }
+
+    client
+  }
+
+  lazy val okHttpClient = newClient
+
+  lazy val okHttpFactory: OkUrlFactory = new OkUrlFactory(okHttpClient)
 }
 
-class HttpHandler extends URLStreamHandlerFactory {
-  def createURLStreamHandler(protocol: String): URLStreamHandler =
-    HttpHandler.okHttpFactory.createURLStreamHandler(protocol)
+class HttpHandler(okHttpFactory: OkUrlFactory = HttpHandler.okHttpFactory) extends URLStreamHandlerFactory {
```

Comment:
This class is [instantiated by reflection](https://github.com/coursier/coursier/blob/dd343267654575ed11265b61de9dc7bf57d14b94/modules/cache/jvm/src/main/scala/coursier/Cache.scala#L225). Are you sure adding an argument doesn't break that?

Reply:
Thanks very much for your comments!

I wasn't aware the OkHttp wasn't the default - I'll certainly look into this and revise the PR. Much appreciated!